src/mpm.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright 2022 The MathWorks, Inc.
+
+import * as core from "@actions/core";
+import * as exec from "@actions/exec";
+import * as tc from "@actions/tool-cache";
+import * as path from "path";
+import properties from "./properties.json";
+
+export async function setup(platform: string, architecture: string): Promise<string> {
+    let mpmUrl: string;
+    if (architecture != "x64") {
+        return Promise.reject(Error(`This action is not supported on ${platform} runners using the ${architecture} architecture.`));
+    }
+    switch (platform) {
+        case "win32":
+            mpmUrl = properties.mpmRootUrl + "win64/mpm";
+            break;
+        case "linux":
+            mpmUrl = properties.mpmRootUrl + "glnxa64/mpm";
+            break;
+        default:
+            return Promise.reject(Error(`This action is not supported on ${platform} runners using the ${architecture} architecture.`));
+    }
+
+    let mpm: string = await tc.downloadTool(mpmUrl);
+    if (platform === "win32") {
+       let mpmExtractedPath: string = await tc.extractZip(mpm);
+       mpm = path.join(mpmExtractedPath, "bin", "win64",  "mpm.exe");
+    }
+
+    const exitCode = await exec.exec(`chmod +x ${mpm}`);
+    if (exitCode !== 0) {
+        return Promise.reject(Error("unable to setup mpm"));
```

Comment:
Is this user-visible?
If so, setup should become "set up" and the message should end with a period.